### DISCUSSION TITLE
Helmfile manager should use specific helmfile version that defined in helmfile.lock when updating helmfile.lock.

### DISCUSSION THREAD

**Discussion by mugioka** [renovatebot/renovate#22699] [Suggest an Idea]
### Tell us more.

Currently, helmfile manager always use latest helmfile version when updating helmfile.lock.
But, if client(ex. Argo CD) does not have compatibility version of helmfile, failed to execute `helmfile template/diff/apply`.
```log
the lockfile was created by Helmfile 0.152.0, which is newer than current 0.151.0; Please upgrade to Helmfile 0.152.0 or greater]
```
So, we should use compatibility helmfile version when updating helmfile.lock.

**Comment by secustor** [comment]
The helmfile manager needs support for tool contraints see as an example: 
- https://github.com/renovatebot/renovate/pull/22244/files#diff-cbe98b2d22cc9a4d857f156b26cfdce5c062af2263b038bd3c3728aeed12cf62R68-R76
- https://github.com/renovatebot/renovate/pull/22456/files#diff-cbe98b2d22cc9a4d857f156b26cfdce5c062af2263b038bd3c3728aeed12cf62R71

**Reply by mugioka** [reply]
the helmfile manager seems to be  already supports tool constraints(https://github.com/renovatebot/renovate/blob/dbb1a41f93d1982db7a9d2ca636824b74293f2db/lib/modules/manager/helmfile/artifacts.ts#L43-L51)
but, that constraints is not enough because of that helmfile version does not set from helmfile.lock.
so, i will be addressed this problem.

**Reply by viceice** [reply]
you currently need to manually configure the constraints inside the renovate config

**Reply by secustor** [reply]
As @viceice mentioned only manual defined constraints are supported currently.

See the second PR for how to implement a constraint detected from a file.

**Reply by mugioka** [reply]
Thank you for reply, i will be make the pull request for that.
